src/bvals/comms/bnd_id.hpp

```diff
@@ -81,6 +81,8 @@ struct BndId {
   BndId() = default;
   KOKKOS_DEFAULTED_FUNCTION
   BndId(const BndId &) = default;
+  KOKKOS_DEFAULTED_FUNCTION
+  ~BndId() = default;
```

Comment:
This appears to be necessary on AMD systems. I'm not sure why we didn't need it until now.